src/main/java/spoon/reflect/declaration/CtTypeInformation.java

```diff
@@ -60,6 +60,11 @@
 	boolean isAnonymous();
 
 	/**
+	 * Return {@code true} if the referenced type is declared in a executable.
+	 */
+	boolean isEnclosingType();
```

Comment:
Then isLocalClass (as in java.lang.Class).

and we can write a nice contract:

for all klass:

  compile(klass).isLocalClass == ctclass(klass).isLocalClass ==
ctclass(klass).reference.isLocalClass